test/function.jl

```diff
@@ -780,6 +780,17 @@ function_form(short::Bool) = string(short ? "short" : "long", "-form")
         end
     end
 
+    @testset "combinedef with no `:head`" begin
+        # should default to `:function`
+        f, expr = @audit function f() end
+
+        d = splitdef(expr)
+        delete!(d, :head)
```

Comment:
Very minor but adds some clarity to the test and will fail if `:head` is still present
```suggestion
        delete!(d, :head)
        @test !haskey(d, :head)
```

Reply:
I don't get it, it confuses me.
Maybe it it was an `@assert`, rather than a `@test` ?
Because it being a `@test` to me implies that we are testing that thing, and that thing is the behavour of `delete!`

Reply:
Either is fine with me. My main point is if I edit this code to remove `delete!(d, :head)` everything will work exactly the same

Reply:
```suggestion
        delete!(d, :head)
        @assert !haskey(d, :head)
```